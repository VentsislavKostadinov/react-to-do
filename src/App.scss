@import './font.scss';

.App {
  font-family: "Poppins-Regular", serif;
  text-align: center;
}