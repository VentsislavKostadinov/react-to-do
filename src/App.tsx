import React, { useState, useEffect } from 'react'
import './App.scss'
import { TaskList } from './TaskList'
import { TaskProps } from './model/TaskProps.types'
import { TaskListProps } from './model/TaskListProps.types'
import { TaskCompletedList } from './TaskListCompleted'

export const App = () => {
    const [tasks, setTasks] = useState(
        JSON.parse(localStorage.getItem('tasks')!) || [],
    )

    useEffect(() => {
        localStorage.setItem('tasks', JSON.stringify(tasks))
    }, [tasks])

    const addTask = (description: string) => {
        if (description) {
            const newTask = {
                id: new Date().getTime(),
                description: description,
                completed: false,
            }
            setTasks([...tasks, newTask])
        }
    }
    const editTask = (taskId: number, newDescription: string) => {
        setTasks((prevTasks: Array<TaskListProps>) =>
            prevTasks.map((task: TaskProps | any) =>
                task.id === taskId
                    ? { ...task, description: newDescription }
                    : task,
            ),
        )
    }
    const deleteTask = (taskId: number) => {
        setTasks((prevTasks: Array<TaskListProps>) =>
            prevTasks.filter((task: TaskProps | any) => task.id !== taskId),
        )
    }

    const completeTask = (taskId: number) => {
        setTasks((prevTasks: Array<TaskListProps>) => {
            return prevTasks.map((task: TaskProps | any) =>
                task.id === taskId
                    ? { ...task, completed: !task.completed }
                    : task,
            )
        })
    }

    const sortIncompletedTasks = (taskList: TaskListProps | any) => {
        return taskList.filter((task: TaskProps['task']) => !task.completed)
    }

    return (
        <div className="to-do">
            <TaskList
                tasks={sortIncompletedTasks(tasks)}
                onAdd={addTask}
                onEdit={editTask}
                onDelete={deleteTask}
                onComplete={completeTask}
            />
            <TaskCompletedList tasks={tasks} />
        </div>
    )
}
