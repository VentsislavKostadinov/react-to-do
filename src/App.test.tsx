import { App } from './App'
import { render } from '@testing-library/react'

describe('App', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<App />)
        expect(asFragment()).toMatchSnapshot()
    })
})
