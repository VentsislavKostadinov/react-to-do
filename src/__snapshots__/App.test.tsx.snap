// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="toDo container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <p
          class="title h2"
        >
          To Do App
        </p>
        <div
          class="inputGroup input-group has-validation"
        >
          <label
            class="form-label"
          />
          <input
            class="searchTask form-control"
            data-testid="search-task"
            placeholder="Search task"
            required=""
            type="search"
            value=""
          />
        </div>
        <div
          class="container container"
        >
          <div
            class="row"
          >
            <div
              class="col col"
            >
              <form
                class=""
                novalidate=""
              >
                <div
                  class="inputGroup input-group has-validation"
                >
                  <label
                    class="form-label"
                  />
                  <input
                    class="addInput form-control"
                    data-testid="add-input"
                    placeholder="Enter a description"
                    required=""
                    type="text"
                    value=""
                  />
                  <button
                    class="btn btn btn-info"
                    type="submit"
                  >
                    Add
                  </button>
                  <div
                    class="invalid-feedback"
                  >
                    Please type a description
                  </div>
                </div>
              </form>
              <p
                class="title h2"
                data-testid="task-list"
              >
                Task list
              </p>
              <div>
                <p
                  class="title h3"
                  data-testid="empty-task-list"
                >
                  You list is empty
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
